### ISSUE TITLE
An error occurs when adding one custom environment if the 'Use app proxy settings' is applying

### ISSUE THREAD

**Issue by v-xianya** [microsoft/AzureStorageExplorer#4938]
**Storage Explorer Version**: 1.21.1
**Build Number**: 20210917.4
**Branch**: hotfix/1.21.1
**Platform/OS**: Windows 10
**Architecture**: ia32
**How Found**: Exploratory testing
**Regression From**: 1.19.0?

## Steps to Reproduce ##
1. Make sure Fiddler is configured well -> Launch Storage Explorer.
2. Go to 'Settings -> Applications' -> Select 'Use app proxy settings'.
3. Set the URL: 'http://localhost' Port: '8888'.
4. Open the connect dialog -> Select 'Subscription -> Custom Environment:' -> Click 'Manage custom environments...'.
5. Click 'Add' -> Type a valid Environment name -> Type 'https://management.azure.com' to ARM endpoint.
6. Click 'Add' -> Check whether succeed to add. 

## Expected Experience ##
Succeed to add.

## Actual Experience ##
An error 'Unable to retrieve configuration. Please verify that the ARM endpoint is correct' occurs.
![image](https://user-images.githubusercontent.com/41351993/133763987-d91d237e-bdf7-4fcc-858a-4293daa7c937.png)

## Additional Context ##
1. This issue also reproduces when selecting 'Use environment variables'. <Steps as below>

> 1. Fiddler is well configured
> 2. Open a command line prompt -> Navigate to the location path of the Storage Explorer.  
> 3. Type the command 'set HTTP_PROXY=http://localhost:8888'.
> 4. Open Storage Explorer from the command line.
> 5. Select 'Use environment variables'.
> 6. Repeat the above steps <4-6> under **Steps to Reproduce** part.

2. Here is the record:
![proxy](https://user-images.githubusercontent.com/41351993/133763930-8b4f1c0f-3310-4644-843e-d314c67ba429.gif)

**Comment by craxal** [comment]
@v-xianya I'm able to reproduce this. The error that occurs appears to be related to certificates:
```
Error: unable to verify the first certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1497:34)
    at TLSSocket.emit (events.js:315:20)
    at TLSSocket._finishInit (_tls_wrap.js:932:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:706:12)
    at TLSWrap.callbackTrampoline (internal/async_hooks.js:131:14) {code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE', stack: 'Error: unable to verify the first certificate…ckTrampoline (internal/async_hooks.js:131:14)', message: 'unable to verify the first certificate'}
```

I was able work around this issue by disabling the Decrypt HTTPS traffic option (Tools > Options... > HTTPS).

I don't think this is something we need to fix unless customers using real proxies start reporting the same issue.

**Comment by v-xianya** [comment]
Fixed on the hotfix build 20210918.4.
